### ISSUE TITLE
Language modeling examples do not show how to do multi-gpu training / fine-tuning

### ISSUE THREAD

**Issue by csiefer2** [huggingface/transformers#31323]
### System Info

- `transformers` version: 4.41.2
- Platform: Linux-5.15.0-1042-nvidia-x86_64-with-glibc2.35
- Python version: 3.9.18
- Huggingface_hub version: 0.23.3
- Safetensors version: 0.4.2
- Accelerate version: 0.31.0
- Accelerate config: 	not found
- PyTorch version (GPU?): 2.2.1+cu121 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed

### Who can help?

@muellerz @stevhliu 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

n/a

### Expected behavior

The `run_clm.py` and other related scripts in:

`https://github.com/huggingface/transformers/tree/main/examples/pytorch/language-modeling`

notionally support training / fine-tuning of models whose gradients are too large to fit on a single GPU, if you believe their CLI.  However there is no example showing how to actually do that.

For instance, `accelerate estimate-memory` says training the Mistral-7B family with Adam takes roughly 55 GB with float16, which is more memory than a single 40GB A100 has.  So I'd need to use more than one GPU.

Would it be possible to modify the language_modeling documentation to explain how to do that?

**Comment by amyeroberts** [comment]
Hi @csiefer2, thanks for opening this issue! 

@muellerzr and @stevhliu are best placed to comment on this in general. 

In the meantime, you can find some accelerate docs on distributed training here: https://huggingface.co/docs/transformers/en/accelerate#distributed-training-with--accelerate

**Comment by muellerzr** [comment]
Just launch the scripts with `accelerate launch` or `torchrun`, no need to do anything else

**Comment by amyeroberts** [comment]
@csiefer2 I can't comment on the memory calculation from accelerate (cc @muellerzr here) but I'm assuming it this is just for the weights of the model + gradient on the forward/backward pass? You'll also need to account for the memory requirements of loading the data onto the GPU. What batch size are you using?

**Comment by stevhliu** [comment]
Thanks for the feedback!

I think keeping these two topics (language modeling and distributed training) in separate docs is better. It sounds like the issue is more about setting up for distributed training, and not so much language modeling. But we can improve on the distributed training docs with an example use case featuring language modeling.

**Comment by stevhliu** [comment]
Thanks for your patience! Working on redesigning the docs right now at https://github.com/huggingface/transformers/pull/31757 and I'll update the distributed training docs when I reach it 🙂
